pmda/density.py

```diff
@@ -0,0 +1,158 @@
+'''
+
+'''
+
+from __future__ import absolute_import
+import numpy as np
+import MDAnalysis as mda
+from MDAnalysis.lib.util import fixedwidth_bins
+from MDAnalysis.analysis.density import Density
+from .parallel import ParallelAnalysisBase
+
+
+class DensityAnalysis(ParallelAnalysisBase):
+    """Parallel density analysis.
+
+    Parameters
+    ----------
+    atomgroup : AtomGroup
+            Group of atoms (such as all the water oxygen atoms) being analyzed.
+            If this is an updating AtomGroup then you need to set
+            'atomselection' and also 'updating=True'.
+    atomselection : str (optional)
+            selection string (MDAnalysis syntax) for the species to be analyzed
+            ["name OH2"]
+    delta : float (optional)
+            bin size for the density grid in Angstroem (same in x,y,z) [1.0]
+            Slice the trajectory as "trajectory[start:stop:step]"; default
+            is to read the whole trajectory.
+    metadata : dict. (optional)
+            `dict` of additional data to be saved with the object; the meta
+            data are passed through as they are.
+    padding : float (optional)
+            increase histogram dimensions by padding (on top of initial box
+            size) in Angstroem. Padding is ignored when setting a user defined
+            grid. [2.0]
+    updating : bool (optional)
+            Should the selection of atoms be updated for every step? ["False"]
+            - "True": atom selection is updated for each frame, can be slow
+            - "False": atoms are only selected at the beginning
+    parameters : dict (optional)
+            `dict` with some special parameters for :class:`Density` (see docs)
+    gridcenter : numpy ndarray, float32 (optional)
+            3 element numpy array detailing the x, y and z coordinates of the
+            center of a user defined grid box in Angstroem ["None"]
+    xdim : float (optional)
+            User defined x dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    ydim : float (optional)
+            User defined y dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    zdim : float (optional)
+            User defined z dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    """
+    def __init__(self, atomgroup, delta=1.0, atomselection=None,
+                 metadata=None, padding=2.0, updating=False,
+                 parameters=None, gridcenter=None, xdim=None, ydim=None,
+                 zdim=None):
```

Comment:
@orbeckst Shall we disable the pylint error `too-many-arguments` for the `__init__` function? It seems pylint doesn't like more than 10 arguments.

Reply:
yes!

Reply:
remove the line https://github.com/MDAnalysis/pmda/blob/29e98369e76cdb596d688a83a8a6b0add9dada38/.pylintrc#L224